skbio/stats/composition.py

```diff
@@ -2058,3 +2061,407 @@ def dirmult_ttest(
+        linear mixed effects models calculated across all of the posterior draws.
+
+        ``qvalue`` is the corrected *p*-value of the linear mixed effects model
+        for multiple comparisons. The reported values are the average of all of
+        the *q*-values computed from the linear mixed effects models calculated
+        across all of the posterior draws.
+
+    See Also
+    --------
+    statsmodels.formula.api.mixedlm
+    statsmodels.regression.mixed_linear_model.MixedLM
+    differential.regression.mixedlm
+
+    Examples
+    --------
+    >>> table = pd.DataFrame(
+    ...     {
+    ...         "u1": [1.00000053, 6.09924644],
+    ...         "u2": [0.99999843, 7.0000045],
+    ...         "u3": [1.09999884, 8.08474053],
+    ...         "x1": [1.09999758, 1.10000349],
+    ...         "x2": [0.99999902, 2.00000027],
+    ...         "x3": [1.09999862, 2.99998318],
+    ...         "y1": [1.00000084, 2.10001257],
+    ...         "y2": [0.9999991, 3.09998418],
+    ...         "y3": [0.99999899, 3.9999742],
+    ...         "z1": [1.10000124, 5.0001796],
+    ...         "z2": [1.00000053, 6.09924644],
+    ...         "z3": [1.10000173, 6.99693644],
+    ...     },
+    ...     index=["Y1", "Y2"],
+    ... ).T
+    >>> metadata = pd.DataFrame(
+    ...     {
+    ...         "patient": [1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4],
+    ...         "treatment": [1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2],
+    ...         "time": [1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3],
+    ...     },
+    ...     index=["x1", "x2", "x3", "y1", "y2", "y3",
+    ...            "z1", "z2", "z3", "u1", "u2", "u3"],
+    ... )
+    >>> res = dirmult_lme(
+    ...     formula="time + treatment", data=table, metadata=metadata,
+    ...     groups="patient", seed=0
+    ... )
+    >>> res
+      FeatureID  Covariate  Log2(FC)   CI(2.5)  CI(97.5)    pvalue    qvalue
+    0        Y1       time -0.210769 -0.731153  0.309650  0.403737  0.403737
+    1        Y1  treatment -0.744093 -1.780094  0.291999  0.252057  0.252057
+    2        Y2       time  0.210769 -0.309650  0.731153  0.403737  0.403737
+    3        Y2  treatment  0.744093 -0.291999  1.780094  0.252057  0.252057
+
+    """
+
+    if not hasattr(data, "__getitem__"):
+        raise ValueError(
+            """data must be a pandas DataFrame or a numpy structured
+              or rec array or a dictionary"""
+        )
+
+    rng = get_rng(seed)
+
+    posterior = [
+        rng.dirichlet(data.values[i] + pseudocount) for i in range(data.shape[0])
+    ]
+
+    dir_table = pd.DataFrame(clr(posterior), index=data.index, columns=data.columns)
+
+    _call_res = _lme_call(
+        formula=formula,
+        table=dir_table,
+        metadata=metadata,
+        groups=groups,
+        reml=reml,
+        method=method,
+        fit_kwargs=fit_kwargs,
+        **kwargs,
+    )
+
+    res = _call_res[0]
+    list_of_vars = _call_res[2]
+
+    submodels = _call_res[1]
+    for i in submodels:
```

Comment:
Note for the future (not in the PR), we will want to have the option to offload this to either joblib or multiprocessing so that models can be independently fitted